src/crystal/system/unix/pthread.cr

```diff
@@ -112,6 +112,22 @@ module Crystal::System::Thread
 
     address
   end
+
+  private def system_name=(name : String?)
+    return unless name
+
+    {% if flag?(:darwin) %}
+      LibC.pthread_setname_np(name)
+    {% elsif flag?(:netbsd) %}
+      LibC.pthread_setname_np(@system_handle, name, nil)
+    {% elsif LibC.has_method?(:pthread_setname_np) %}
+      LibC.pthread_setname_np(@system_handle, name)
+    {% elsif LibC.has_method?(:pthread_set_name_np) %}
+      LibC.pthread_set_name_np(@system_handle, name)
+    {% end %}
```

Comment:
**suggestion:** To make sure we're not missing anything, add a raising `else` branch. We're doing this in similar cases already.

```suggestion
    {% else %}
      {% raise "No `Crystal::System::Thread#system_name` implementation available" %}
    {% end %}
```

Reply:
I'm fine with it, but the feature is far from critical, so failing compilation seems harsh. This is also why I'm not checking the return types either.

For example: that would fail compilation on Android where the feature won't work (you must go through JNI to name the thread) so I skipped the LibC definition on purpose.

Reply:
Hm. But is it fine to just ignore the request to set the thread name then?

Reply:
`int pthread_setname_np(pthread_t __pthread, const char* __name)` exists on Bionic. It is only `int pthread_getname_np(pthread_t __pthread, char* __buf, size_t __n)` that doesn't exist until API level 26 (but we have to support 24)

Reply:
Apparently we must go through JNI for naming threads (in 2018 at least):
https://stackoverflow.com/questions/50327631/android-ndk-setting-a-name-for-a-c-thread

There's no harm in calling it though.

Reply:
It might not update the name on the JVM side, but as far as native code is concerned, I can confirm this works with LLDB on Termux

Reply:
![Screenshot_20240129-201111~2.png](https://github.com/crystal-lang/crystal/assets/1361918/2742f075-38ff-4b10-8da9-46d66bc32aa8)